### DISCUSSION TITLE
Data visualization similar to hex workshop.

### DISCUSSION THREAD

**Discussion by Uzver123** [WerWolv/ImHex#1554] [Get Help] [answered]
There is a [data visualizer](http://www.hexworkshop.com/onlinehelp/650/html/Data_Visualizer.htm) in hex workshop that shows map for each line of hex code in the nearby window located next to editor. However i think its best to have this feature apply directly to the hex code the editor. 

Is there way to define color ranges according to decimal value of each byte?

```
The segmented palette displays bands of colors based on the byte value of the data.
[ 0 - 31]  Grey scale; ASCII Control characters
[ 32 - 127 ] Green; ASCII characters
[ 127 - 254 ]  Blue; Extended ASCII characters
[ 255 ]  Black
```

**Comment by WerWolv** [comment, marked as answer]
ImHex has a minimap since a few commits ago if that's what you mean. Not yet in a release but it's available in the nightly builds.

It works line-by-line and not byte-by-byte though.
<img width="783" alt="image" src="https://github.com/WerWolv/ImHex/assets/10835354/b84d7f9e-6cb6-438f-b8ea-9dc981e909a9">

If you want to do highlighting of the hex cells directly, look into Highlighting Rules. `Edit -> Modify highlighting rules...`

<img width="855" alt="image" src="https://github.com/WerWolv/ImHex/assets/10835354/d01dd8cf-99f1-4020-9a2b-c99bb3d82d83">
